e2e/src/bin/historic_trades.rs

```diff
@@ -34,38 +34,46 @@ fn main() -> Result<()> {
     let mut report = Report::new(File::create(options.output_dir.join("trades.csv"))?);
     report.header()?;
 
-    cmd::for_each_batch_sync(&options.orderbook_file, |history, batch| {
-        let auction_elements = history.auction_elements_for_batch(batch)?;
-        let settlement = history.settlement_for_batch(batch);
-
-        let new_orders = auction_elements
-            .iter()
-            .filter(|order| batch == order.valid.from)
-            .filter(|order| order.price.numerator != 0 && order.price.denominator > MIN_AMOUNT)
-            .filter(|order| order.balance > U256::from(MIN_AMOUNT));
-        for order in new_orders {
-            let settlement = settlement.as_ref();
-            let pricegraph = Pricegraph::new(
-                auction_elements
-                    .iter()
-                    .filter(|element| (element.user, element.id) != (order.user, order.id))
-                    .cloned(),
-            );
-
-            let meta = OrderMetadata::compute(settlement, order, &pricegraph);
-            let result = if meta.is_reasonably_priced_order() {
-                process_reasonable_order(&meta)
-            } else {
-                process_unreasonable_order(&meta)
-            };
-
-            report.record_result(batch, settlement, order, &meta, result)?;
-        }
-
-        Ok(())
-    })?;
+    cmd::for_each_batch(
+        &options.orderbook_file,
+        report,
+        |sampler, history, batch| {
+            let auction_elements = history.auction_elements_for_batch(batch)?;
+            let settlement = history.settlement_for_batch(batch);
+
+            let new_orders = auction_elements
+                .iter()
+                .filter(|order| batch == order.valid.from)
+                .filter(|order| order.price.numerator != 0 && order.price.denominator > MIN_AMOUNT)
+                .filter(|order| order.balance > U256::from(MIN_AMOUNT));
+            for order in new_orders {
+                let settlement = settlement.as_ref();
+                let pricegraph = Pricegraph::new(
+                    auction_elements
+                        .iter()
+                        .filter(|element| (element.user, element.id) != (order.user, order.id))
+                        .cloned(),
+                );
+
+                let meta = OrderMetadata::compute(settlement, order, &pricegraph);
+                let result = if meta.is_reasonably_priced_order() {
+                    process_reasonable_order(&meta)
+                } else {
+                    process_unreasonable_order(&meta)
+                };
+
+                sampler.sample(Sample {
```

Comment:
This line is a tongue-twister. Why is the `sampler` not called `reporter`, isn't it the same thing?

Reply:
:see_no_evil: 

Yes, maybe the `Sampler` struct should be renamed. Its used for sending samples to the main thread in order to be processed by the `reporter`, so its a little different.